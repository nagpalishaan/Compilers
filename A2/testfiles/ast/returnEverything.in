int n() { 
    a = 0;
    if (b == 0) {
        return a;
    } else {
        return 1;
    }
    return function(a, b);
}