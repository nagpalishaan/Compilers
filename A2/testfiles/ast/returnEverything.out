Program
  List
    Function
      IdDecl ID="n"
      Opt
      Opt
        List
          AssignVar
            IdDecl ID="a"
            Numeral NUMERAL="0"
          IfElse
            Equal
              IdUse ID="b"
              Numeral NUMERAL="0"
            Opt
              List
                Return
                  IdUse ID="a"
            Opt
              List
                Return
                  Numeral NUMERAL="1"
          Return
            FunctionCall
              IdUse ID="function"
              Opt
                List
                  IdUse ID="a"
                  Mod
                    IdUse ID="b"
                    IdUse ID="a"
