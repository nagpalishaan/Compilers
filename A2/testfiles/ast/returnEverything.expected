Program
  List
    Function
      IdDecl ID="n"
      Opt
      Opt
        List
          AssignVar
            IdDecl ID="a"
            Numeral NUMERAL="0"
          IfElse
            Equal
              IdUse ID="b"
              Numeral NUMERAL="0"
            List
              Return
                IdUse ID="a"
            Opt
              List
                Return
                  Numeral NUMERAL="1"
          FunctionCallStmt
            FunctionCall
              IdDecl ID="function"
              Opt
                List
                  IdUse ID="a"
                  Mod
                    IdUse ID="b"
                    IdUse ID="a"
